src/resources/js/events-admin.js

```diff
@@ -364,7 +364,7 @@ jQuery( function( $ ) {
 			group.append( fields );
 		} );
 
-		section.on( 'click', '.tribe-delete-this', function(e) {
+		$( document ).on( 'click', `#event_${post_type} .tribe-delete-this`, function(e) {
```

Comment:
This binds the delete-this action to the document, so any new delete-this button innately has the action tied to it.